luigi/worker.py

```diff
@@ -108,10 +108,10 @@ def __init__(self, task, worker_id, result_queue, tracking_url_callback,
         self.result_queue = result_queue
         self.tracking_url_callback = tracking_url_callback
         self.status_message_callback = status_message_callback
-        self.random_seed = random_seed
         if task.worker_timeout is not None:
             worker_timeout = task.worker_timeout
         self.timeout_time = time.time() + worker_timeout if worker_timeout else None
+        self.random_seed = random_seed or self.timeout_time is not None
```

Comment:
It won't be set if you're running a task that doesn't have a timeout on 1 worker.